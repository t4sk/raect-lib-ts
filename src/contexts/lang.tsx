import React, { useState, createContext, useContext, useMemo } from "react";

const DEFAULT_LANG = "en";

const LangContext = createContext({
  state: DEFAULT_LANG,
  setLang: (_lang: string) => {},
});

export function useLangContext() {
  return useContext(LangContext);
}

interface ProviderProps {}

export const Provider: React.FC<ProviderProps> = ({ children }) => {
  const [state, setState] = useState(DEFAULT_LANG);

  function setLang(lang: string) {
    setState(lang);
  }

  return (
    <LangContext.Provider
      value={useMemo(
        () => ({
          state,
          setLang,
        }),
        [state]
      )}
    >
      {children}
    </LangContext.Provider>
  );
};
