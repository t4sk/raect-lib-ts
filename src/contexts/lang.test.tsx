import React from "react";
import { shallow } from "enzyme";
import { Provider, useLangContext } from "./lang";

interface Props {}

const TestComponent: React.FC<Props> = (props) => {
  const { state } = useLangContext();

  return <div>{state}</div>;
};

test("it renders", () => {
  const component = shallow(
    <Provider>
      <TestComponent />
    </Provider>
  );

  expect(component).toMatchSnapshot();
});
