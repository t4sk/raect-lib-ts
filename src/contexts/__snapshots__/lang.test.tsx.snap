// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<ContextProvider
  value={
    Object {
      "setLang": [Function],
      "state": "en",
    }
  }
>
  <TestComponent />
</ContextProvider>
`;
